Luke7787/ImmaculateInventors
# Feature/create account backend

Summary:
- Updated user schema to accommodate for the CreateAccount front-end fields
- Added validation to verify that all fields are filled out before creating a new user
- Connected backend to frontend

How to test:
- in express-backend, `npm run dev`
- Open boomerang
- POST to localhost:8000/users/
   - `{
      "username": "dlog",
      "password": "a1234567890!AA",
      "email": "adam@gmail.com",
      "country": "United States",
      "state": "California",
      "city": "San Diego",
      "zipcode": "93410",
      "firstName": "Adam",

      }`
      - Verify that you receive an error saying **lastName** is missing
      - Repeat this for every field, each time leaving a different field unfilled

- POST to localhost:8000/users/
    - `{
      "username": "dlog",
      "password": "a1234567890!AA",
      "email": "adam@gmail.com",
      "country": "United States",
      "state": "California",
      "city": "San Diego",
      "zipcode": "93410",
      "firstName": "Adam",
      "lastName": "Ta"
      }`
      - Verify that it is successful in adding the user by GET request to localhost:8000/users

- in root, `npm start`
- click sign in and then create account
- open chrome dev tools
- hit "Submit"
   - Verify that field validation appears and that nothing is logged in console
- Fill out every field with validation passing
   - Verify that the response is console logged
   - Go to boomerang
   - GET to localhost:8000/users/
       - Verify that the user you just created on the front-end is visible in the backend
